### PR TITLE
[NS-377] Automate schema generation for canvas-data by leveraging schema API

### PR DESCRIPTION
**Pull request by sandeepmjay**
https://jira.ets.berkeley.edu/jira/browse/NS-377

### CODE REVIEW

File: nessie/jobs/refresh_canvas_data_catalog.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+
+"""Logic for generate canvas data catalog job."""
+
+
+class RefreshCanvasDataCatalog(BackgroundJob):
+
+    def run(self):
+        # Retrieve latest schema definitions from Canvas data API
+        response = canvas_data.get_canvas_data_schema(mock=None)
```
Comment (paulkerschen) [inline nessie/jobs/refresh_canvas_data_catalog.py]: @sandeepmjay It should be safe to leave `mock=None` out of this line; that argument is only something we'd hook into during a test.

File: nessie/jobs/refresh_canvas_data_catalog.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+            external_table_ddl = '{}\n{}{}{}{};\n\n'.format(
+                external_table_ddl,
+                create_ddl,
+                storage_descriptors,
+                table_properties,
+                table_location,
+            )
+
+        # For debugging prposes export to external_table_ddl to file to get a well formed SQL template for canvas-data
```
Comment (paulkerschen) [inline nessie/jobs/refresh_canvas_data_catalog.py]: Typo `prposes`

**Comment by paulkerschen** [comment]
@sandeepmjay Left a couple of minor comments, but thanks much for doing this; it will be very nice to have automated schema creation on hand.

**Comment by sandeepmjay** [comment]
Thanks, @paulkerschen. Amends are in.

**Pull request merged**